File: valid-parentheses 13-04-2024.js
var isValid = function(s) {
    let dictionary = {
        ')' : '(',
        '}' : '{',
        ']' : '['
    };
    let stack = [];
    for (let c of s) {
        if (dictionary[c]) {
            if (stack.length && stack[stack.length - 1] === dictionary[c]) {
                stack.pop();
            } else {
                return false;
            }
        } else {
            stack.push(c);
        }
    }

    return !stack.length;
};

File: valid-parentheses 13-04-2024.py
def isValid(s):
        dictionary = {')': '(', '}': '{', ']': '['}
        stack = []
        for c in s:
            if c in dictionary:
                if stack and stack[-1] == dictionary[c]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(c)
        return True if not stack else False